terraform/modules/happy-cloudfront/variables.tf

```diff
@@ -0,0 +1,67 @@
+variable "frontend" {
+  type = object({
+    domain_name = string
+    zone_id     = string
+  })
+  description = "The domain name and zone ID the user will see."
+}
+
+variable "origin" {
+  type = object({
+    domain_name = string
+  })
+  description = "The domain name of the origin."
+}
+
+variable "cache" {
+  type = object({
+    min_ttl     = optional(number, 0)
+    default_ttl = optional(number, 300)
+    max_ttl     = optional(number, 300)
+    compress    = optional(bool, true)
+  })
+  description = "The cache settings for the CloudFront distribution."
+  default     = {}
+}
+
+variable "geo_restriction_locations" {
+  type        = set(string)
+  default     = ["US", "CA", "MX"]
+  description = "The countries to whitelist for the CloudFront distribution."
+}
+
+variable "allowed_methods" {
+  type        = set(string)
+  default     = ["GET", "HEAD"]
```

Comment:
I think we might want to add POST and PUT?

Reply:
I should change this variable name. It is only related to cacheing allowed methods. All methods are allowed and forwarded